commands/scale_cluster.go

```diff
@@ -38,108 +38,277 @@ Examples:
+	}
 
-	if err == nil {
-		return
+	// We have to marshal and unmarshal here because the generated client gives us
+	// a map[string]interface and we want to unmarshal it into the actual
+	// apiextensions type.
+	m, err := json.Marshal(response.Payload)
+	if err != nil {
+		return nil, err
 	}
 
-	handleCommonErrors(err)
+	var status v1alpha1.StatusCluster
 
-	// print non-common error
-	fmt.Println(color.RedString(err.Error()))
-	os.Exit(1)
+	err = json.Unmarshal(m, &status)
+	if err != nil {
+		return nil, err
+	}
+
+	return &status, nil
 }
 
-// verifyScaleClusterPreconditions does a few general checks and returns an error in case something is missing.
-func verifyScaleClusterPreconditions(args scaleClusterArguments, cmdLineArgs []string) error {
-	if config.Config.Token == "" && args.authToken == "" {
-		return microerror.Mask(notLoggedInError)
+// scaleCluster is the actual function submitting the API call and handling the response.
+func scaleCluster(args scaleClusterArguments) error {
+	// Preparing API call.
+	reqBody := &models.V4ModifyClusterRequest{
+		Scaling: &models.V4ModifyClusterRequestScaling{
+			Max: args.workersMax,
+			Min: args.workersMin,
+		},
 	}
-	if len(cmdLineArgs) == 0 {
-		return microerror.Mask(clusterIDMissingError)
+
+	// perform API call
+	if args.verbose {
+		fmt.Println("Sending API request to modify cluster")
+	}
+
+	auxParams := ClientV2.DefaultAuxiliaryParams()
+	auxParams.ActivityName = scaleClusterActivityName
+
+	_, err := ClientV2.ModifyCluster(args.clusterID, reqBody, auxParams)
+	if err != nil {
+		if clientErr, ok := err.(*clienterror.APIError); ok {
+			if clientErr.HTTPStatusCode == http.StatusForbidden {
+				return microerror.Mask(accessForbiddenError)
+			} else if clientErr.HTTPStatusCode == http.StatusNotFound {
+				return microerror.Mask(clusterNotFoundError)
+			}
+		}
+
+		return microerror.Mask(err)
 	}
+
 	return nil
 }
 
 // scaleClusterRunOutput invokes the actual cluster scaling and prints the result and/or errors.
 func scaleClusterRunOutput(cmd *cobra.Command, cmdLineArgs []string) {
-	args := defaultScaleClusterArguments()
-	args.clusterID = cmdLineArgs[0]
+	if len(cmdLineArgs) == 0 {
+		handleCommonErrors(clusterIDMissingError)
+	}
+	// Get all necessary information for defaulting and confirmation.
+	clusterDetails, err := getClusterDetails(cmdLineArgs[0], scaleClusterActivityName)
+	if err != nil {
+		fmt.Println(color.RedString("Error getting cluster details!"))
+		handleCommonErrors(err)
+		fmt.Println(color.RedString(err.Error()))
+		os.Exit(1)
+	}
+	status, err := getClusterStatus(cmdLineArgs[0], scaleClusterActivityName)
+	if err != nil {
+		fmt.Println(color.RedString("Error getting cluster status!"))
+		handleCommonErrors(err)
+		fmt.Println(color.RedString(err.Error()))
+		os.Exit(1)
+	}
+
+	var maxBefore int64
+	var minBefore int64
+	var desiredCapacity int64
+	// Older clusters without scaling shouldn't have scaling values but have a workers array instead.
```

Comment:
I decided against checking the cluster version here and instead checking if there are scaling settings present. Is this okay to do or should I do the semver comparison instead here?

Reply:
I think the version checks would be the more proper way of doing it. Same for what we do in the API.

Reply:
I added a version check like in the API.